src/index.js

```diff
@@ -5,72 +5,135 @@ const addButton = document.getElementById('button');
-    todoList.removeChild(li);
-  };
+const savedTasks = JSON.parse(localStorage.getItem('tasks')) || [];
 
-  const editTask = (event) => {
-    const li = event.target.parentNode;
-    const taskValue = li.firstChild.textContent.trim();
-    const newTaskValue = prompt('Edit task:', taskValue);
-    if (newTaskValue !== null && newTaskValue.trim() !== '') {
-      li.firstChild.textContent = newTaskValue.trim();
-    } else {
-      return;
-    }
-
-    const popupDiv = document.createElement('div');
-    popupDiv.classList.add('popup');
-    const heading = document.createElement('h2');
-    heading.textContent = 'Edit task';
-    const input = document.createElement('input');
-    input.type = 'text';
-    input.value = newTaskValue.trim();
-    const saveButton = document.createElement('button');
-    saveButton.textContent = 'Save';
-    saveButton.addEventListener('click', () => {
-      const editedTaskValue = input.value.trim();
-      if (editedTaskValue !== '') {
-        li.firstChild.textContent = editedTaskValue;
-        popupDiv.remove();
-      }
-    });
-    popupDiv.appendChild(heading);
-    popupDiv.appendChild(input);
-    popupDiv.appendChild(saveButton);
-    document.body.appendChild(popupDiv);
-  };
-
-  const addTask = () => {
-    const taskValue = taskInput.value.trim();
+/* eslint-disable */
 
+function renderTasks() {
+  savedTasks.forEach((task) => {
     const li = document.createElement('li');
     const checkbox = document.createElement('input');
     checkbox.type = 'checkbox';
+    checkbox.checked = task.completed;
     li.appendChild(checkbox);
-    li.innerHTML += `${taskValue} <button class="remove-button">X</button> <button class="edit-button">Edit</button>`;
+    li.innerHTML += `${task.text} <button class="remove-button">X</button> <button class="edit-button">Edit</button>`;
     todoList.appendChild(li);
 
-    taskInput.value = '';
     const removeButton = li.querySelector('.remove-button');
-    const editButton = li.querySelector('.edit-button');
-    removeButton.addEventListener('click', removeTask);
-    editButton.addEventListener('click', editTask);
-  };
+    removeButton.addEventListener('click', () => {
+      removeTask(li);
+    });
 
-  const removeCompletedTasks = () => {
-    const checkboxes = document.querySelectorAll('input[type="checkbox"]');
-    checkboxes.forEach((checkbox) => {
-      if (checkbox.checked) {
-        const li = checkbox.parentNode;
-        todoList.removeChild(li);
-      }
+    const editButton = li.querySelector('.edit-button');
+    editButton.addEventListener('click', () => {
+      editTask(li);
     });
+  });
+}
+renderTasks(); // edite
+
+// save tassk
+
+function saveTasks() {
+  localStorage.setItem('tasks', JSON.stringify(savedTasks));
+}
+
+function addTask() {
+  const taskValue = taskInput.value.trim();
+
+  if (taskValue === '') {
+    return;
+  }
+
+  const task = {
+    text: taskValue,
+    completed: false,
   };
```

Comment:
![image](https://user-images.githubusercontent.com/93039359/232188106-681bbe21-bee0-445f-88de-2e7a5f24afb9.png)

- [ ] As the previous reviewer suggested changes about the index value :+1: Kindly fix all the changes which you can see in the image above :+1:
- [ ] Secondly, it would be great if you add an index in the task object because your todo object is gonna take three key-pair values **(description, completed, index)** as required in this project :+1: